content/docs/code-splitting.md

```diff
@@ -0,0 +1,205 @@
+    // `chunkFilename` provides a template for naming code-split bundles (optional)
+    chunkFilename: '[name].bundle.js',
+    // `path` is the folder where Webpack will place your bundles
+    path: './dist',
+    // `publicPath` is where Webpack will load your bundles from (optional)
+    publicPath: 'dist/'
+  }
+};
+```
+
+When using [Babel](http://babeljs.io/) you'll need to make sure that Babel can
+parse the dynamic import syntax but is not transforming it. For that you will need [babel-plugin-syntax-dynamic-import](https://yarnpkg.com/en/package/babel-plugin-syntax-dynamic-import).
+
+## Libraries
+
+### React Loadable
+
+[React Loadable](https://github.com/thejameskyle/react-loadable) provides you
+with a nice API for introducing code splitting into your app where-ever you
+render a component.
+
+**Before:**
+
+```js
+import OtherComponent from './OtherComponent';
+
+const MyComponent = () => (
+  <OtherComponent/>
+);
+```
+
+**After:**
+
+```js
+import Loadable from 'react-loadable';
+
+const LoadableOtherComponent = Loadable({
+  loader: () => import('./OtherComponent'),
+  loading: () => <div>Loading...</div>,
+});
+
+const MyComponent = () => (
+  <LoadableOtherComponent/>
+);
+```
+
+React Loadable helps you rendering
+[loading states](https://github.com/thejameskyle/react-loadable#creating-a-great-loading-component),
+[error states](https://github.com/thejameskyle/react-loadable#loading-error-states),
+[timeouts](https://github.com/thejameskyle/react-loadable#timing-out-when-the-loader-is-taking-too-long),
+[preloading](https://github.com/thejameskyle/react-loadable#preloading), and
+more. It can even help you [server-side](https://github.com/thejameskyle/react-loadable#------------server-side-rendering)
+render an app with lots of code-splitting.
+
+## Route-based code splitting
+
+Deciding where in your app to introduce code splitting can be a bit tricky. You
+want to make sure you choose places that will split bundles evenly, but not be
+terribly interuptive to the users of your app.
+
+A good place is to start is with routes. Most people on the web are used to
+page transitions taking some amount of time to load. You also tend to be
+re-rendering the entire page at once so your users are unlikely to be
+interacting with other elements on the page at the same time.
+
+Here's an example of how to setup route-based code splitting into your app using
+libraries like [React Router](https://reacttraining.com/react-router/) and
+[React Loadable](https://github.com/thejameskyle/react-loadable).
+
+```js
+import { BrowserRouter as Router, Route } from 'react-router-dom';
+import Loadable from 'react-loadable';
+
+const Loading = () => <div>Loading...</div>;
+
+const Home = Loadable({
+  loader: () => import('./routes/Home'),
+  loading: Loading,
+});
+
+const About = Loadable({
+  loader: () => import('./routes/About'),
+  loading: Loading,
+});
+
+const App = () => (
+  <Router>
+    <Route path="/" component={Home}/>
```

Comment:
We're going to fix this soon, but right now this route config won't give you what you probably expect because `path="/"` does *not* try to match the entire URL pathname, so `Home` will actually render at the `/about` URL as well. This behavior was intended to make it easier for people to nest routes, but it backfires in the `/` case 🤦‍♂️  You can read more about it [here](https://github.com/ReactTraining/react-router/issues/4958).

Anyway, the fix for this example would be to either a) use a more specific `path` (e.g. `path="/home"`) or b) use `<Route exact path="/" .../>`, which will match the URL exactly.

Reply:
whoops, thanks